src/main/java/nwoolcan/model/brewery/warehouse/Warehouse.java

```diff
@@ -94,5 +95,30 @@ IngredientArticle createIngredientArticle(String name,
      * @return a {@link Result} of {@link Article} for fluency.
      */
     Result<Article> setName(Article article, String newName);
+    /**
+     * Getter of a single {@link Article} based on its id.
+     * @param id to search the {@link Article}.
+     * @return the {@link Article} requested or an error if it does not exist.
+     */
+    Result<Article> getArticleById(int id);
+    /**
+     * Getter of a single {@link Article} based on its id and on its type.
+     * @param id to search the {@link Article}.
+     * @param articleType the type of the article.
+     * @return the {@link Article} requested or an error if it does not exist.
+     */
+    Result<Article> getArticleById(ArticleType articleType, int id);
```

Comment:
How is this different from requesting only the id?

Reply:
it requires to be a precise articletype, so that you can be sure you can cast it. It should be equivalend to make three different methods:
1) getBeerArticleById
2) getMiscArticleById
3) getIngredientArticlyById

Reply:
But we do have methods to cast safely an article into Beer/Misc/IngredientArticle

Reply:
> Still, if you query something with this method and it returns a valuedResult, you can be sure to use those methods.
Makes sense, you should probably use a more understandable error then if the id is present but it's not of the specified type, something like "Wrong article type" instead of `ARTICLE_NOT_FOUND`

Reply:
I didn't get your point. that method actually is useless. I removed it